core/src/main/java/org/infinispan/cache/impl/CacheImpl.java

```diff
@@ -381,11 +378,7 @@ public final boolean isEmpty() {
    }
 
    final boolean isEmpty(EnumSet<Flag> explicitFlags, ClassLoader explicitClassLoader) {
-      // TODO: replace with stream findAny isPresent instead
-      try (CloseableIterable<CacheEntry<K, Void>> iterable = filterEntries(AcceptAllKeyValueFilter.getInstance(),
-            explicitFlags, explicitClassLoader).converter(NullValueConverter.getInstance())) {
-         return !iterable.iterator().hasNext();
-      }
+      return !entrySet(explicitFlags, explicitClassLoader).stream().anyMatch(StreamMarshalling.alwaysTruePredicate());
```

Comment:
I would have expected to see `entrySet(explicitFlags, explicitClassLoader).isEmpty()` here, but I see it's the other way around - ` `cache.isEmpty()`, perhaps ?